nibabel/externals/netcdf.py

```diff
@@ -607,7 +607,10 @@ def _read_var_array(self):
                     pos = self.fp.tell()
                     self.fp.seek(begin_)
                     data = fromstring(self.fp.read(a_size), dtype=dtype_)
-                    data.shape = shape
+                    # Avoid exception for empty variables.
+                    if len(data.shape) != 1 or data.shape[0] != 0:
+                        data.shape = shape
```

Comment:
The only case I've seen is the "empty" netcdf case with zero bytes allocated. I prefer the minimal change, happy to simplify the test.